src/server/scripts/Zandalar/Underrot/boss_sporecaller_zancha.cpp

```diff
@@ -0,0 +1,707 @@
+        AfterEffectRemove += AuraEffectRemoveFn(spell_sporecaller_zancha_upheaval_marker::OnRemove, EFFECT_0, SPELL_AURA_DUMMY, AURA_EFFECT_HANDLE_REAL);
+    }
+};
+
+// 272511 - Upheaval
+class spell_sporecaller_zancha_upheaval_script_effect : public SpellScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_UPHEAVAL_MISSILE });
+    }
+
+    void HandleHitTarget(SpellEffIndex /*effIndex*/) const
+    {
+        GetHitUnit()->CastSpell(GetCaster(), SPELL_UPHEAVAL_MISSILE, TRIGGERED_IGNORE_CAST_TIME);
+    }
+
+    void Register() override
+    {
+        OnEffectHitTarget += SpellEffectFn(spell_sporecaller_zancha_upheaval_script_effect::HandleHitTarget, EFFECT_0, SPELL_EFFECT_SCRIPT_EFFECT);
+    }
+};
+
+// 259888 - Festering Harvest
+class spell_sporecaller_zancha_festering_harvest : public AuraScript
+{
+    static constexpr std::array<uint8, 5> SporecallerEnergizeCycle = { 2, 2, 2, 2, 3 };
+
+    void PeriodicTick(AuraEffect const* aurEff) const
+    {
+        uint8 cycleIdx = aurEff->GetTickNumber() % SporecallerEnergizeCycle.size();
+        GetTarget()->ModifyPower(POWER_ENERGY, SporecallerEnergizeCycle[cycleIdx]);
+    }
+
+    void Register() override
+    {
+        OnEffectPeriodic += AuraEffectPeriodicFn(spell_sporecaller_zancha_festering_harvest::PeriodicTick, EFFECT_0, SPELL_AURA_PERIODIC_DUMMY);
+    }
+};
+
+// 259732 - Festering Harvest
+class spell_sporecaller_zancha_festering_harvest_pods_selector : public SpellScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_SPORE_POD_BEAM });
+    }
+
+    void OnPrecast() override
+    {
+        std::list<Creature*> sporePods;
+        Trinity::AllCreaturesOfEntryInRange checker(GetCaster(), NPC_SPORE_POD, 200.0f);
+        Trinity::CreatureListSearcher<Trinity::AllCreaturesOfEntryInRange> searcher(GetCaster(), sporePods, checker);
+        Cell::VisitAllObjects(GetCaster(), searcher, 200.0f);
+
+        if (sporePods.empty())
+            return;
+
+        for (Creature* spore : sporePods)
+        {
+            spore->CastSpell(spore, SPELL_SPORE_POD_BEAM, CastSpellExtraArgsInit{
+                .TriggerFlags = TRIGGERED_IGNORE_CAST_IN_PROGRESS | TRIGGERED_DONT_REPORT_CAST_ERROR,
+                .TriggeringSpell = GetSpell()
+            });
+        }
+    }
+
+    void Register() override { }
+};
+
+// 259845 - Boundless Rot
+class spell_sporecaller_zancha_boundless_rot : public SpellScript
+{
+    static void SetDest(SpellDestination& dest)
+    {
+        dest.Relocate(CenterPosition);
+    }
+
+    void Register() override
+    {
+        OnDestinationTargetSelect += SpellDestinationTargetSelectFn(spell_sporecaller_zancha_boundless_rot::SetDest, EFFECT_0, TARGET_DEST_DEST);
+    }
+};
+
+// 273271 - Volatile Pods
+class spell_sporecaller_zancha_volatile_pods : public SpellScript
+{
+    void HandleHitTarget(SpellEffIndex /*effIndex*/) const
```

Comment:
I would mark them all parameters SpellEffIndex as const?

```suggestion
    void HandleHitTarget(SpellEffIndex const /*effIndex*/) const
```